XenAdmin/Diagnostics/Problems/HostProblem/HostMaintenanceMode.cs

```diff
@@ -63,8 +63,12 @@ public override string HelpMessage
 
         public override AsyncAction CreateUnwindChangesAction()
         {
-            Program.MainWindow.CloseActiveWizards(Server.Connection);
-            return new DisableHostAction(Server);
+            var server = Server.Connection.Resolve(new XenRef<Host>(Server.opaque_ref));
+            if (server == null)
+                return null;
+
+            Program.MainWindow.CloseActiveWizards(server.Connection);

```

Comment:
One more thing that I didn't notice the first time I read the code: I think we want to close the active wizards on all occasions even if we can't resolve the server, in which case this line should move above.

Reply:
I understand, I will move it above the Resolve line.